src/main/java/lanat/argumentTypes/TryParseArgument.java

```diff
@@ -42,44 +42,45 @@ protected void addError(@NotNull String value) {
 
 	private @Nullable Function<String, Object> getParseMethod() {
 		// Get a static valueOf(String), a parse(String), or a from(String) method.
-		final var method = Arrays.stream(this.type.getMethods())
-			.filter(m -> Modifier.isStatic(m.getModifiers()))
+		final var method = this.type.getMethods()
+			.filter(MMethod::isStatic)
 			.filter(this::isValidMethod)
 			.filter(m -> (m.getName().equals("valueOf") || m.getName().equals("from") || m.getName().equals("parse")))
 			.findFirst();
 
 		// if we found a method, return that.
 		if (method.isPresent()) {
-			return s -> {
+			return input -> {
 				try {
-					return method.get().invoke(null, s);
-				} catch (IllegalAccessException | InvocationTargetException e) {
-					this.addError(s);
+					return method.get().invoke(input);
+				} catch (MirrorException exception) {
+					this.addError(input);
 				}
 				return null;
 			};
 		}
 
 		// Otherwise, try to find a constructor that takes a string.
-		return Arrays.stream(this.type.getConstructors())
+		return this.type.getConstructors()
 			.filter(this::isValidMethod)
 			.findFirst()
 			.<Function<String, Object>>map(constructor -> s -> {
 				try {
-					return constructor.newInstance(s);
-				} catch (InstantiationException | IllegalAccessException | InvocationTargetException e) {
+					return constructor.invoke(s);
+				} catch (MirrorException exception) {
 					this.addError(s);
 				}
 				return null;
 			}).orElse(null);
 	}
 
 	@Override
+	@SuppressWarnings("unchecked")
 	public @Nullable T parseValues(@NotNull String @NotNull [] args) {
 		try {
-			return this.type.cast(this.parseMethod.apply(args[0]));
+			return (T) this.parseMethod.apply(args[0]);
```

Comment:
```suggestion
			return (T)this.parseMethod.apply(args[0]);
```
Remove the space after the casting operator :trollface: